packages/@lwc/engine-core/src/framework/modules/computed-style-attr.ts

```diff
@@ -4,21 +4,31 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import { isNull, isString } from '@lwc/shared';
+import { isNull, isString, isUndefined } from '@lwc/shared';
 import { RendererAPI } from '../renderer';
-import { VBaseElement } from '../vnodes';
+import { VBaseElement, VStaticPart } from '../vnodes';
+import { logError } from '../../shared/logger';
+import { VM } from '../vm';
 
 // The style property is a string when defined via an expression in the template.
 export function patchStyleAttribute(
-    oldVnode: VBaseElement | null,
-    vnode: VBaseElement,
-    renderer: RendererAPI
+    oldVnode: VBaseElement | VStaticPart | null,
+    vnode: VBaseElement | VStaticPart,
+    renderer: RendererAPI,
+    owner: VM
 ) {
     const {
         elm,
         data: { style: newStyle },
     } = vnode;
 
+    if (!(isNull(newStyle) || isUndefined(newStyle)) && !isString(newStyle)) {
+        logError(
+            `Invalid 'style' attribute passed to <${elm!.tagName.toLowerCase()}> is ignored. This attribute must be a string value.`,
+            owner
+        );
+    }
```

Comment:
This `logError` should be wrapped in a `process.env.NODE_ENV !== 'production'` check.